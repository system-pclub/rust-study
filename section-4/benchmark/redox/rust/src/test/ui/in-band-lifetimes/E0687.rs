#![allow(warnings)]
#![feature(in_band_lifetimes)]

fn foo(x: fn(&'a u32)) {} //~ ERROR must be explicitly

fn bar(x: &Fn(&'a u32)) {} //~ ERROR must be explicitly

fn baz(x: fn(&'a u32), y: &'a u32) {} //~ ERROR must be explicitly

struct Foo<'a> { x: &'a u32 }

impl Foo<'a> {
    fn bar(&self, x: fn(&'a u32)) {} //~ ERROR must be explicitly
}

fn main() {}
