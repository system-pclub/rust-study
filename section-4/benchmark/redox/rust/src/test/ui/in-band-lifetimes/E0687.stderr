error[E0687]: lifetimes used in `fn` or `Fn` syntax must be explicitly declared using `<...>` binders
  --> $DIR/E0687.rs:4:15
   |
LL | fn foo(x: fn(&'a u32)) {}
   |               ^^ in-band lifetime definition

error[E0687]: lifetimes used in `fn` or `Fn` syntax must be explicitly declared using `<...>` binders
  --> $DIR/E0687.rs:6:16
   |
LL | fn bar(x: &Fn(&'a u32)) {}
   |                ^^ in-band lifetime definition

error[E0687]: lifetimes used in `fn` or `Fn` syntax must be explicitly declared using `<...>` binders
  --> $DIR/E0687.rs:8:15
   |
LL | fn baz(x: fn(&'a u32), y: &'a u32) {}
   |               ^^ in-band lifetime definition

error[E0687]: lifetimes used in `fn` or `Fn` syntax must be explicitly declared using `<...>` binders
  --> $DIR/E0687.rs:13:26
   |
LL |     fn bar(&self, x: fn(&'a u32)) {}
   |                          ^^ in-band lifetime definition

error: aborting due to 4 previous errors

